source/joker_effects.c

```diff
@@ -488,87 +558,140 @@ static JokerEffect the_trio_joker_effect(Joker *joker, Card *scored_card) {
     return effect;
  }
 
-static JokerEffect the_family_joker_effect(Joker *joker, Card *scored_card) {
+static JokerEffect the_family_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
     
-     // This is really inefficient but the only way at the moment to check for whole-hand conditions
+    // This is really inefficient but the only way at the moment to check for whole-hand conditions
     u8 suits[NUM_SUITS];
     u8 ranks[NUM_RANKS];
     get_played_distribution(ranks, suits);
 
-    if (hand_contains_n_of_a_kind(ranks) >= 4)
+    if (hand_contains_n_of_a_kind(ranks) >= 4) {
         effect.xmult = 4;
+    }
+
     return effect;
  }
 
-static JokerEffect the_order_joker_effect(Joker *joker, Card *scored_card) {
+
+static JokerEffect the_order_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
     JokerEffect effect = {0};
-    if (scored_card != NULL)
-        return effect; // if card != null, we are not at the end-phase of scoring yet
+
+    SCORE_INDEPENDANT_ONLY(scored_card, scored_when, effect)
```

Comment:
```suggestion
    SCORE_ON_EVENT_ONLY(JOKER_CALLBACK_INDEPENDANT, scored_when, effect)
```

Batch - Consolidate joker event restriction macros